### PR TITLE
 [inductor] Replace self_cuda_time_total function calls with self_dev…

### PR DESCRIPTION
**Pull request by peaceorwell**
…ice_time_total for wrapper_bench

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by ezyang** [comment]
Could you explain why this is an improvement?

**Comment by peaceorwell** [comment]
> Could you explain why this is an improvement?

@ezyang  Before the modification, the code here only worked on CUDA devices. After the modification, the code in this file can now be directly reused by non-CUDA related devices. Additionally, self_cuda_time_total is a deprecated API, and it's recommended to use the more general self_device_time_total interface instead. Thanks！

File: torch/_inductor/wrapper_benchmark.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def get_self_cuda_time(ev):
         """
         ev.self_cuda_time_total is in microsecond. Convert to millisecond.
```
Comment (shunting314) [inline torch/_inductor/wrapper_benchmark.py:159]: Can you update this comment accordingly?

**Review by shunting314 — APPROVED** [review]

**Comment by shunting314** [comment]
BTW, I found `wrapper_benchmark.py` contains other occurrences of `self_cuda_time_total`. It would be better that we replace all of them to `self_device_time_total`

**Comment by shunting314** [comment]
If you have a GPU supporting triton, can you make sure:
```
python <wrapper>.py -p
```
still works? Refer to this doc for reference: https://pytorch.org/docs/stable/torch.compiler_inductor_profiling.html

**Comment by peaceorwell** [comment]
> If you have a GPU supporting triton, can you make sure:
> 
> ```
> python <wrapper>.py -p
> ```
> 
> still works? Refer to this doc for reference: https://pytorch.org/docs/stable/torch.compiler_inductor_profiling.html

@shunting314 
Based on the current modifications, run the following script. 
```
/tmp/torchinductor_zhouyusong/ig/cigg7qvo374dat7mlnukqjtve25wwjsfayjej4untisjqryntcqo.py -p
```

The output log is as follows:
```
0.000039
/home/zhouyusong/venv_main/lib/python3.10/site-packages/torch/_inductor/runtime/triton_heuristics.py:852: UserWarning: Unable to infer type of value for keyword: stream (Triggered internally at /home/zhouyusong/pytorch_main/torch/csrc/profiler/python/init.cpp:257.)
  with torch._C._profiler._RecordFunctionFast(
0.000070
Profiling result for a compiled module of benchmark None:
Chrome trace for the profile is written to /tmp/compiled_module_profile.json
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  --------------------------------------------------------  
                                                   Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg      Self MLU    Self MLU %     MLU total  MLU time avg    # of Calls                                              Input Shapes  
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  --------------------------------------------------------  
                                 triton_poi_fused_sub_0        43.97%       2.061ms        53.49%       2.507ms      25.070us       1.784ms        96.30%       1.784ms      17.842us           100      [[128, 4, 4096], [128, 4, 4096], [128, 4, 4096], []]  
                                 triton_poi_fused_sub_0         0.00%       0.000us         0.00%       0.000us       0.000us       1.784ms        96.30%       1.784ms      17.842us           100                                                        []  
                                          aten::normal_        21.37%       1.002ms        23.74%       1.113ms     556.272us      68.640us         3.70%      68.640us      34.320us             2                                   [[2097152], [], [], []]  
MLURandomNormalKernelPhilox(void*, unsigned long, fl...         0.00%       0.000us         0.00%       0.000us       0.000us      68.640us         3.70%      68.640us      34.320us             2                                                        []  
                                            aten::randn         1.12%      52.405us        34.25%       1.605ms     802.670us       0.000us         0.00%      68.640us      34.320us             2                                      [[], [], [], [], []]  
                                            aten::empty         9.58%     449.036us         9.58%     449.036us     112.259us       0.000us         0.00%       0.000us       0.000us             4                                  [[], [], [], [], [], []]  
                           cnnlRandGetSimulateThreadNum         0.05%       2.244us         0.05%       2.244us       1.122us       0.000us         0.00%       0.000us       0.000us             2                                                        []  
                                 cnnlGenerateRandNormal         0.35%      16.370us         2.32%     108.702us      54.351us       0.000us         0.00%       0.000us       0.000us             2                                                        []  
                                         cnInvokeKernel        11.49%     538.420us        11.49%     538.420us       5.279us       0.000us         0.00%       0.000us       0.000us           102                                                        []  
                                       aten::as_strided         0.34%      15.823us         0.34%      15.823us       7.911us       0.000us         0.00%       0.000us       0.000us             2                                   [[2097152], [], [], []]  
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  --------------------------------------------------------  
Self CPU time total: 4.687ms
Self MLU time total: 1.853ms


  == triton_pointwise category kernels == 
Kernel                    Self MLU TIME (ms)  Count    Percent
----------------------  --------------------  -------  ---------
triton_poi_fused_sub_0             0.0178416  1.0      45.93%
Total                              0.0178416           45.93%

  == unknown category kernels == 
Kernel                                                                                                                     Self MLU TIME (ms)  Count    Percent
-----------------------------------------------------------------------------------------------------------------------  --------------------  -------  ---------
MLURandomNormalKernelPhilox(void*, unsigned long, float, float, unsigned int, unsigned int, unsigned int, unsigned int,             0.0006864  0.02     1.77%
Total                                                                                                                               0.0006864           1.77%

Percent of time when MLU is busy: 47.70%
Total wall time 0.039 ms
Output for tabulate: None, 45.93%, 0.00%, 0.00%, 0.00%, 1.77%, 47.70%, 0.039ms
```

**Review by shunting314 — APPROVED** [review]

**Comment by peaceorwell** [comment]
@shunting314 Hi, is there anything else that needs to be modified?

**Comment by shunting314** [comment]
It looks good. Can you merge it?

**Comment by peaceorwell** [comment]
> It looks good. Can you merge it?

@shunting314 This PR is currently in the "workflows awaiting approval" status. You may need to approve and run the workflows.​​​​​​​​​

**Comment by peaceorwell** [comment]
@shunting314 I fixed a lint error. Could you please trigger the pipeline again? Thank you.

**Comment by peaceorwell** [comment]
@shunting314 Could you help me add a "ciflow/inductor" label?

**Pull request closed without merging**